### ISSUE TITLE
Check

### ISSUE THREAD

**Issue by vaibhavkhamitkar12** [vaibhavkhamitkar12/Resume#1]
import pandas as pd
from rai_insights import RAIInsights
from raiutils.models import wrap_model
from raiwidgets import ResponsibleAIDashboard
from sklearn.metrics import f1_score
import numpy as np

# Step 1: Sample DataFrame
# Replace this with your actual data
data = {
    'text': [
        'Apple is looking at buying U.K. startup for $1 billion',
        'San Francisco considers banning sidewalk delivery robots',
        'London is a big city in the United Kingdom.'
    ],
    'predicted_entities': [
        ['Apple', 'U.K.'],
        ['San Francisco'],
        ['London', 'United Kingdom']
    ],
    'true_entities': [
        ['Apple', 'U.K.'],
        ['San Francisco'],
        ['London', 'United Kingdom']
    ],
    'category': ['Business', 'Policy', 'Geography']  # Optional for cohort/fairness analysis
}

df = pd.DataFrame(data)

# Step 2: Convert list of entities to set and create an accuracy metric column
def jaccard_similarity(pred, true):
    return len(set(pred).intersection(set(true))) / len(set(pred).union(set(true)))

df['jaccard_score'] = [
    jaccard_similarity(p, t) for p, t in zip(df['predicted_entities'], df['true_entities'])
]

# Step 3: Create a binary label for perfect match
df['exact_match'] = df['jaccard_score'].apply(lambda x: 1 if x == 1.0 else 0)

# Step 4: Wrap model (simulate classification by checking if prediction was exact)
def predict_fn(data):
    return data['exact_match']

# Use wrap_model to create a model-like wrapper
wrapped_model = wrap_model(predict_fn)

# Step 5: RAIInsights initialization
rai_insights = RAIInsights(
    model=wrapped_model,
    train=df,
    test=df,
    target_column='exact_match',
    task_type='classification'
)

# Step 6: Add Responsible AI components
rai_insights.error_analysis.add()
rai_insights.fairness.add(sensitive_features=['category'])  # Optional
rai_insights.compute()

# Step 7: Launch Dashboard
ResponsibleAIDashboard(rai_insights)
